tests/test_user_util.py

```diff
@@ -36,3 +20,39 @@ def test_command_line_interface():
     help_result = runner.invoke(cli.main, ['--help'])
     assert help_result.exit_code == 0
     assert '--help  Show this message and exit.' in help_result.output
+
+
+@pytest.mark.parametrize(
+    'salt', (
```

Comment:
I'm guessing that making salt an iterable is so that we can rotate it, but then still be able to check usernames hashed with the previous salts? So every time we add one it will increase the number of things we need to check every time a user is created? If so I think we'll need "get_all_username/email_hashes" so that we can get any one that may have been generated throughout history and check against all of them in one go.

Reply:
Correct on all counts. I'll add a generator for that.

Reply:
Added.